### PR TITLE
Fix Failure if Namespace is in Terminating State

### PR DESCRIPTION
**Pull request by jonathan-innis**
Previously, the executor would fail to execute steps if the namespace is in a `Terminating` state; now, we perform retry on our API server actions so that we don't fail if the first call fails

This should also resolve the issue that I was seeing in testing

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,10 +26,15 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+const (
+	HelmReleaseGroup = "helm.toolkit.fluxcd.io"
+	HelmReleaseKind  = "HelmRelease"
+	DefaultRetryTime = time.Second * 10
```
Comment (nitishm) [inline main.go:32]: If naming it DefaultRetryTime maybe we should give the end user, either a cli flag or an env variable to override this? WDYT?

Comment (jonathan-innis) [reply main.go:32]: In my mind this is business logic, is there a reason that we would want this to be configurable?

Comment (nitishm) [reply main.go:30]: let's make it lower case in that case?

Comment (jonathan-innis) [reply main.go:30]: Now that I consider it, it's not that much work to pass this up, I'll pass this as a parameter through the argo DAG step

**Review by nitishm — APPROVED** [review]
🚀

**Pull request merged**